frontend/src/container/withActivity.jsx

```diff
@@ -119,28 +120,65 @@ const withActivity = (WrappedComponent, setActivityList, setActivityNextPage, re
       }
     }
 
+    /**
+     * Get a complete comment
+     *
+     * This function exists also in appContentFactory
+     * @param {int} spaceId
+     * @param {int} contentId
+     * @param {int} commentId
+     * @returns {Promise<JSON>}
+     */
+    getComment = async (spaceId, contentId, commentId) => {
+      const { props } = this
+      const fetchGetComment = await handleFetchResult(await getComment(FETCH_CONFIG.apiUrl, spaceId, contentId, commentId))
+      switch (fetchGetComment.apiResponse.status) {
+        case 200: return fetchGetComment.body
+        default:
+          props.dispatch(newFlashMessage(props.t('Unknown comment')))
+          return {}
+      }
+    }
+
+    /**
+     * Update the activity list according to the TLM
+     * Every operation on ActivityList from TLM should be in this function
+     * @param {TLM} data
+     * @returns
+     */
     updateActivityListFromTlm = async (data) => {
       const { props } = this
-      if (
-        data.event_type === `${TLM_ET.CONTENT}.${TLM_CET.MODIFIED}.${TLM_SUB.COMMENT}` ||
-        data.event_type === `${TLM_ET.CONTENT}.${TLM_CET.DELETED}.${TLM_SUB.COMMENT}`
-      ) return
       await this.waitForNoChange()
       this.changingActivityList = true
       let activity = data
-      if (
-        (data.event_type.includes(TLM_ET.CONTENT) && !(
-          data.event_type.includes(TLM_SUB.COMMENT) || data.event_type.includes(TLM_SUB.TODO)
-        )) ||
-        (data.event_type.includes(TLM_ET.MENTION) && data.fields.content.content_type !== TLM_SUB.COMMENT)
-      ) {
+      if (data.event_type.includes(TLM_SUB.COMMENT)) {
         activity = {
           ...data,
           fields: {
             ...data.fields,
             content: {
               ...data.fields.content,
-              ...await this.getContent(data.fields.content.content_id)
+              ...await this.getComment(
+                data.fields.workspace.workspace_id,
+                data.fields.content.parent_id,
+                data.fields.content.content_id
+              )
+            }
+          }
+        }
+      } else {
+        if (
+          (data.event_type.includes(TLM_ET.CONTENT) && !data.event_type.includes(TLM_SUB.TODO)) ||
+          data.event_type.includes(TLM_ET.MENTION)
```

Comment:
is it all right that here it is the same logic for mentions in notes and mentions in comments?

Reply:
Correct me if I'm wrong, but I don't think it's the same logic
If the sub type is a comment, then I enter in the first part of the condition.
Then, if it's not a comment, we do theses lines (170-171-172). Therefore, if it's a mention in a comment, the executed code is lines 154 to 168

Reply:
Sorry, my answer is maybe late but

> If the sub type is a comment, then I enter in the first part of the condition.

The TLM type will be only `mention.created`. The information about what type of content the mention was made in will be inside the TLM object.

Reply:
Fixed in a40c65b

Reply:
:warning: Since we got two TLM (one for the creation of the comment, one for the mention) we will fetch twice the comment